dbclient/health/src/main/java/io/helidon/dbclient/health/DbClientHealthCheck.java

```diff
@@ -58,39 +72,204 @@ public static Builder builder(DbClient dbClient) {
+        protected Awaitable<?> execPing() {
+            return dbClient().execute(exec -> exec.namedDml(statementName));
+        }
+
+    }
+
+    /**
+     * Database health check which calls DBClient's {@code dml} method.
+     */
+    private static final class DbClientHealthCheckAsDml extends DbClientHealthCheck {
+
+        /** Custom statement. */
+        private final String statement;
+
+        private DbClientHealthCheckAsDml(Builder builder) {
+            super(builder);
+            this.statement = builder.statement;
+            LOGGER.log(Level.FINEST, "Created an instance of DbClientHealthCheckAsDml");
+        }
+
+        @Override
+        protected Awaitable<?> execPing() {
+            return dbClient().execute(exec -> exec.dml(statement));
+        }
+
+    }
+
+    /**
+     * Database health check which calls DBClient's {@code namedQuery} method.
+     */
+    private static final class DbClientHealthCheckAsNamedQuery extends DbClientHealthCheck {
+
+        /** Name of the statement. */
+        private final String statementName;
+
+        private DbClientHealthCheckAsNamedQuery(Builder builder) {
+            super(builder);
+            this.statementName = builder.statementName;
+            LOGGER.log(Level.FINEST, "Created an instance of DbClientHealthCheckAsNamedQuery");
+        }
+
+        @Override
+        protected Awaitable<?> execPing() {
+            return dbClient()
+                    .execute(exec -> exec.namedQuery(statementName).forEach(it -> {}));
+        }
+
+    }
+
+    /**
+     * Database health check which calls DBClient's {@code query} method.
+     */
+    private static final class DbClientHealthCheckAsQuery extends DbClientHealthCheck {
+
+        /** Custom statement. */
+        private final String statement;
+
+        private DbClientHealthCheckAsQuery(Builder builder) {
+            super(builder);
+            this.statement = builder.statement;
+            LOGGER.log(Level.FINEST, "Created an instance of DbClientHealthCheckAsQuery");
+        }
+
+        @Override
+        protected Awaitable<?> execPing() {
+            return dbClient()
+                    .execute(exec -> exec.query(statement).forEach(it -> {}));
+        }
+
+    }
+
     /**
      * Fluent API builder for {@link DbClientHealthCheck}.
+     * Default health check setup will call named DML statement with name {@code ping}.
+     * This named DML statement shall be configured in {@code statements} section
+     * of the DBClient configuration file.
      */
     public static final class Builder implements io.helidon.common.Builder<DbClientHealthCheck> {
+
+        /** Helidon database client. */
         private final DbClient database;
+        /** Health check name. */
         private String name;
+        /** Health check timeout in seconds (to wait for statement execution response). */
+        private int timeoutSeconds;
+
+        // Those two boolean variables define 4 ways of query execution:
```

Comment:
Maybe a multiline comment would be a better fit?

Reply:
timeoutSeconds was replaced with timeoutDuration/timeoutUnit and comments updated.